native/packages/react-native-bpk-component-button/src/BpkButton.android.js

```diff
@@ -76,66 +72,70 @@ const BpkButton = (props: Props) => {
-  // have been supplied. If not, disable theming.
-  if (theme) {
-    if (!isTypeThemeable(type) || !themeAttributesSupplied(type, theme)) {
-      theme = null;
-    }
+  const shouldApplyTheme =
+    themeProp &&
+    (isTypeThemeable(type) && themeAttributesSupplied(type, themeProp));
+  const theme = shouldApplyTheme ? themeProp : null;
+
+  const accessibilityTraits = ['button'];
+  const containerStyle = getStyleForElement('container', props);
+  const buttonStyle = getStyleForElement('button', props);
+  const backgroundColor = getAndroidBackgroundColour(theme, props);
+
+  if (disabled) {
+    accessibilityTraits.push('disabled');
   }
 
-  const renderIcon = (): ?Node => {
-    if (!icon) {
-      return null;
-    }
-    if (typeof icon === 'string') {
-      return (
-        <BpkIcon icon={icon} style={iconStyle(theme, props)} small={!large} />
-      );
-    }
-    return icon;
-  };
   return (
-    <BpkButtonContainer
-      accessibilityLabel={accessibilityLabel}
-      disabled={disabled}
-      icon={icon}
-      iconOnly={iconOnly}
-      onPress={onPress}
-      style={style}
-      theme={theme}
-      title={title}
-      type={type}
-      large={large}
-      {...rest}
-    >
-      <View style={getStyleForElement('view', props)}>
-        {!iconOnly && (
-          <BpkText
-            textStyle={large ? 'lg' : 'sm'}
-            emphasize
-            style={textStyle(theme, props)}
-          >
-            {Platform.OS === 'android' ? title.toUpperCase() : title}
-          </BpkText>
-        )}
-        {renderIcon()}
-      </View>
-    </BpkButtonContainer>
+    <View style={[containerStyle, style]}>
+      <TouchableNativeFeedback
+        disabled={disabled}
+        onPress={onPress}
+        style={style}
+        type={type}
+        accessibilityComponentType="button"
+        accessibilityLabel={accessibilityLabel || title}
+        accessibilityTraits={accessibilityTraits}
+        background={TouchableNativeFeedback.SelectableBackgroundBorderless()}
+        {...rest}
+      >
+        <View style={[buttonStyle, backgroundColor]}>
+          <View style={getStyleForElement('view', props)}>
+            {!iconOnly && (
+              <BpkText
+                textStyle={large ? 'lg' : 'sm'}
+                emphasize
+                style={textStyle(theme, props)}
+              >
+                {title}
+              </BpkText>
+            )}
+            {typeof icon === 'string' ? (
+              <BpkIcon
+                icon={icon}
+                style={iconStyle(theme, props)}
+                small={!large}
+              />
+            ) : (
+              icon
+            )}
+          </View>
+        </View>
+      </TouchableNativeFeedback>
+    </View>
   );
 };
 
-const propTypes = {
+BpkButton.propTypes = {
   ...COMMON_PROP_TYPES,
   iconAlignment: PropTypes.oneOf(Object.keys(ICON_ALIGNMENTS)),
   large: PropTypes.bool,
```

Comment:
`iconAlignment` and `large` are common so can both be abstracted.

Reply:
`large` isn't required on Android (readme should reflect this too).

Reply:
Im gonna do these in a follow up PR - I promise 🙏